opta/core/aws.py

```diff
@@ -167,6 +161,35 @@ def force_delete_terraform_lock_id(self) -> None:
             Key={"LockID": {"S": f"{bucket}/{self.layer.name}"}},
         )
 
+    @classmethod
+    def get_all_remote_configs(cls) -> Dict[str, Dict[str, "StructuredConfig"]]:
+        prefix = "opta_config/"
+        s3 = boto3.client("s3")
+        opta_config_detailed_map = {}
+        for aws_bucket in cls._get_bucket_list():
+            detailed_configs = {}
+            response = s3.list_objects(Bucket=aws_bucket, Prefix=prefix, Delimiter="/")
+            if "Contents" in response:
+                for data in response["Contents"]:
+                    structured_config = cls._download_remote_config(
+                        s3, aws_bucket, data["Key"]
+                    )
+                    if structured_config:
+                        detailed_configs[data["Key"][len(prefix) :]] = structured_config
+                opta_config_detailed_map[aws_bucket] = detailed_configs
+        return opta_config_detailed_map
+
+    @staticmethod
+    def _get_bucket_list() -> List[str]:
```

Comment:
this method is also doing some filtering, so the name should capture this information `_list_opta_buckets`